docs/spec/lightclient/Lightclient_A_1.tla

```diff
@@ -0,0 +1,463 @@
+    IF ~ValidAndVerifiedPre(trusted, untrusted)
+    THEN "FAILED_VERIFICATION"
+    ELSE IF ~BC!InTrustingPeriod(untrusted.header)
+    THEN "FAILED_TRUSTING_PERIOD" 
+    ELSE IF untrusted.header.height = trusted.header.height + 1
+             \/ OneThird(trusted, untrusted)
+         THEN "OK"
+         ELSE "CANNOT_VERIFY"
+
+(*
+ Initial states of the light client.
+ Initially, only the trusted light block is present.
+ *)
+LCInit ==
+    /\ state = "working"
+    /\ nextHeight = TARGET_HEIGHT
+    /\ nprobes = 0  \* no tests have been done so far
+    /\ LET trustedBlock == blockchain[TRUSTED_HEIGHT]
+           trustedLightBlock == [header |-> trustedBlock, Commits |-> AllNodes]
+       IN
+        \* initially, fetchedLightBlocks is a function of one element, i.e., TRUSTED_HEIGHT
+        /\ fetchedLightBlocks = [h \in {TRUSTED_HEIGHT} |-> trustedLightBlock]
+        \* initially, lightBlockStatus is a function of one element, i.e., TRUSTED_HEIGHT
+        /\ lightBlockStatus = [h \in {TRUSTED_HEIGHT} |-> "StateVerified"]
+        \* the latest verified block the the trusted block
+        /\ latestVerified = trustedLightBlock
+
+\* copy the block from the light store        
+LightStoreGetInto(block, height) ==
+    block = fetchedLightBlocks[height]
+
+\* block should contain a copy of the block from the reference chain, with a matching commit
+CopyLightBlockFromChain(block, height) ==
+    LET ref == blockchain[height]
+        lastCommit ==
+          IF height < ULTIMATE_HEIGHT
+          THEN blockchain[height + 1].lastCommit
+            \* for the ultimate block, which we never use, as ULTIMATE_HEIGHT = TARGET_HEIGHT + 1
+          ELSE blockchain[height].VS 
+    IN
+    block = [header |-> ref, Commits |-> lastCommit]      
+
+\* Either the primary is correct and the block comes from the reference chain,
+\* or the block is produced by a faulty primary.
+\*
+\* TODO: add a traceability tag when Josef adds it in the English spec
+FetchLightBlockInto(block, height) ==
+    IF IS_PRIMARY_CORRECT
+    THEN CopyLightBlockFromChain(block, height)
+    ELSE BC!IsLightBlockAllowedByDigitalSignatures(height, block)
+
+\* add a block into the light store    
+\*
+\* TODO: add a traceability tag when Josef adds it in the English spec
+LightStoreUpdateBlocks(lightBlocks, block) ==
+    LET ht == block.header.height IN    
+    [h \in DOMAIN lightBlocks \union {ht} |->
+        IF h = ht THEN block ELSE lightBlocks[h]]
+
+\* update the state of a light block      
+\*
+\* TODO: add a traceability tag when Josef adds it in the English spec
+LightStoreUpdateStates(statuses, ht, blockState) ==
+    [h \in DOMAIN statuses \union {ht} |->
+        IF h = ht THEN blockState ELSE statuses[h]]      
+
+\* Check, whether newHeight is a possible next height for the light client.
+\*
+\* TODO: add a traceability tag when Josef adds it in the English spec
+ScheduleTo(newHeight, pNextHeight, pTargetHeight, pLatestVerified) ==
```

Comment:
Regarding the name, "schedule" comes from the Rust implementation, so we better keep it like that. Let's rename it to `CanSchedule` though.

Reply:
For the responsibilities of the caller, it is better to keep the verified and block scheduler as independent from each other, as possible. Splitting the decision into two parts on two sides complicates the interface. The current version might be hard to read, but that is because it captures all possible block schedulers we may implement in the future. A concrete implementation of the scheduler should be more intuitive.

Reply:
We have changed the parameter order. Your version makes more sense.